internal/gengapic/custom_operation.go

```diff
@@ -298,6 +338,57 @@ func operationResponseField(m *descriptor.DescriptorProto, target string) *descr
 	return nil
 }
 
+// operationRequestField is a helper for extracting the operation_request_field annotation from a field.
+func operationRequestField(f *descriptor.FieldDescriptorProto) (string, bool) {
+	mapping := proto.GetExtension(f.GetOptions(), extendedops.E_OperationRequestField).(string)
+	if mapping != "" {
+		return mapping, true
+	}
+
+	return "", false
+}
+
+// operationPollingMethod is a helper for finding the operation service RPC annotated with operation_polling_method.
+func operationPollingMethod(s *descriptor.ServiceDescriptorProto) *descriptor.MethodDescriptorProto {
+	for _, m := range s.GetMethod() {
+		if proto.GetExtension(m.GetOptions(), extendedops.E_OperationPollingMethod).(bool) {
+			return m
+		}
+	}
+
+	return nil
+}
+
+// pollingRequestParamters collects the polling request parameters for an operation service's polling method
+// based on which are annotated in an initiating RPC's request message with operation_request_field and that
+// are also marked as required on the polling request message. Specifically, this weeds out the parent_id field
+// of the GlobalOrganizationOperations polling params.
+func (g *generator) pollingRequestParameters(m *descriptor.MethodDescriptorProto, opServ *descriptor.ServiceDescriptorProto) []string {
+	poll := operationPollingMethod(opServ)
+	if poll == nil {
+		return nil
+	}
+	pollReqName := poll.GetInputType()
+
+	inType := g.descInfo.Type[m.GetInputType()].(*descriptor.DescriptorProto)
+
+	params := []string{}
```

Comment:
nit: `var params []string` maybe useful enough as nil slice is meaningful. If this works could move it to the top and always return the slice?

Reply:
SGTM! Done.